brokers/redis/broker.go

```diff
@@ -11,30 +11,45 @@ import (
 )
 
 const (
-	pollPeriod = time.Second
+	DefaultPollPeriod = time.Second
 )
 
 type Options struct {
-	Addrs    []string
-	Password string
-	DB       int
+	Addrs        []string
+	Password     string
+	DB           int
+	DialTimeout  time.Duration
+	ReadTimeout  time.Duration
+	WriteTimeout time.Duration
+	IdleTimeout  time.Duration
+	MinIdleConns int
+	PollPeriod   time.Duration
 }
 
 type Broker struct {
-	log  logf.Logger
-	conn redis.UniversalClient
+	log        logf.Logger
+	conn       redis.UniversalClient
+	pollPeriod time.Duration
 }
 
 func New(o Options, lo logf.Logger) *Broker {
+	pollPeriod := o.PollPeriod
+	if o.PollPeriod == 0 {
+		pollPeriod = DefaultPollPeriod
+	}
 	return &Broker{
 		log: lo,
-		conn: redis.NewClient(
-			&redis.Options{
-				Addr:     o.Addrs[0],
-				Password: o.Password,
-				DB:       o.DB,
-			},
-		),
+		conn: redis.NewUniversalClient(&redis.UniversalOptions{
```

Comment:
Thoughts on passing an existing `redis.UniversalClient` (concurrent safe) instead of creating a new conn pool? It would also support a fail-over client and reduce options verbosity aside from reusing an existing conn pool.